inbm/dispatcher-agent/dispatcher/source/source_command.py

```diff
@@ -113,16 +112,29 @@ def _handle_app_source_command(
         return Result(status=200, message=serialized_list)
 
     if "remove" in app_action:
-        keyname = parsed_head.get_children("applicationSource/remove/gpg")["keyname"]
+        keyname = None
+        try:
+            keyname = parsed_head.get_children("applicationSource/remove/gpg")["keyname"]
+        except XmlException:
+            # These children may not be present
+            logger.info(f"Optional GPG key parameters not present in manifest")
         filename = parsed_head.get_children("applicationSource/remove/repo")["filename"]
         application_source_manager.remove(
             ApplicationRemoveSourceParameters(file_name=filename, gpg_key_name=keyname)
         )
         return Result(status=200, message="SUCCESS")
 
     if "add" in app_action:
-        gpg_key_uri = parsed_head.get_children("applicationSource/add/gpg")["uri"]
-        gpg_key_name = parsed_head.get_children("applicationSource/add/gpg")["keyname"]
+        gpg_key_uri = None
+        gpg_key_name = None
+
+        try:
+            gpg_key_uri = parsed_head.get_children("applicationSource/add/gpg")["uri"]
+            gpg_key_name = parsed_head.get_children("applicationSource/add/gpg")["keyname"]
+        except XmlException:
+            # These children may not be present
+            logger.info(f"Optional GPG key parameters not present in manifest")
+
         repo_filename = parsed_head.get_children("applicationSource/add/repo")["filename"]
```

Comment:
**Suggestion:** Use a context manager to ensure that the file operations are properly managed and the file is closed in case of an error. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        with parsed_head.get_children("applicationSource/add/repo") as repo_info:
            repo_filename = repo_info["filename"]
```

Reply:
done